Framework/Core/src/TableTreeHelpers.cxx

```diff
@@ -85,310 +85,124 @@ class ColumnIterator
 };
 } // namespace
 
-// is used in TableToTree
-BranchIterator::BranchIterator(TTree* tree, std::shared_ptr<arrow::ChunkedArray> col, std::shared_ptr<arrow::Field> field)
+ColumnToBranchBase::ColumnToBranchBase(arrow::ChunkedArray* column, arrow::Field* field, int size)
+  : mBranchName{field->name()},
+    mColumn{column},
+    listSize{size}
 {
-  mField = field.get();
-  mBranchName = mField->name();
-
-  mFieldType = mField->type()->id();
-  mChunks = col->chunks();
-  mNumberChuncs = mChunks.size();
-
-  mLeaflistString = mBranchName;
-  mElementType = mFieldType;
-  mNumberElements = 1;
-  if (mFieldType == arrow::Type::type::FIXED_SIZE_LIST) {
-
-    // element type
-    if (mField->type()->num_fields() <= 0) {
-      LOGP(FATAL, "Field {} of type {} has no children!", mField->name(), mField->type()->ToString().c_str());
-    }
-    mElementType = mField->type()->field(0)->type()->id();
-    // number of elements
-    mNumberElements = static_cast<const arrow::FixedSizeListType*>(mField->type().get())->list_size();
-    mLeaflistString += "[" + std::to_string(mNumberElements) + "]";
-  }
-
-  // initialize the branch
-  mStatus = initBranch(tree);
-
-  mCounterChunk = 0;
-  mStatus &= initDataBuffer(mCounterChunk);
 }
 
-BranchIterator::~BranchIterator()
+TableToTree::TableToTree(std::shared_ptr<arrow::Table> table, TFile* file, const char* treename)
 {
-  delete mBranchBuffer;
-
-  delete[] mVariable_o;
+  mTable = table.get();
+  mTree = static_cast<TTree*>(file->Get(treename));
+  if (mTree == nullptr) {
```

Comment:
```
if (mTree != nullptr) {
  return;
}
```

etc...

Reply:
No, this is supposed to create a new tree in case file does not have it.

Reply:
Sorry, misunderstood at first, fixed now.

Reply:
Yes, that you can have afterwards, but if you already have a tree, there is nothing else that needs to be done.